### PR TITLE
[MM-27271] add MM-T1390: Enforce Guest MFA when MFA is enabled and enforced

### PR DESCRIPTION
**Pull request by isacikgoz**
#### Summary
Adds test case MM-T1390: Enforce Guest MFA when MFA is enabled and enforced.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-27271

### CODE REVIEW

File: e2e/package.json [hunk 1]
```diff
@@ -55,5 +55,8 @@
     "test": "start-test start:webhook 3000 test:ci",
     "test:ci": "node run_tests.js",
     "uniq-meta": "grep -r \"^// $META:\" cypress | grep -ow '@\\w*' | sort | uniq"
+  },
+  "dependencies": {
+    "authenticator": "^1.1.5"
```
Comment (saturninoabril) [inline e2e/package.json:60]: Please use fixed version and move to "devDependencies" as we're putting it all there (no distinction for E2E).

File: e2e/cypress/integration/enterprise/guest_accounts/guest_identification_spec.js [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+import {getEmailUrl, getEmailMessageSeparator, getRandomId} from '../../../utils';
+
+const authenticator = require('authenticator');
+
+describe('Guest Accounts', () => {
+    let sysadmin;
+    let testTeam;
+    let testChannel;
+    let adminMFASecret;
+    const username = 'g' + getRandomId(); // usernames has to starts with a letter.
```
Comment (saturninoabril) [inline e2e/cypress/integration/enterprise/guest_accounts/guest_identification_spec.js:26]:
```suggestion
    const username = 'g' + getRandomId(); // username has to starts with a letter.
```

File: e2e/cypress/integration/enterprise/guest_accounts/guest_identification_spec.js [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+        cy.apiRequireLicenseForFeature('GuestAccounts');
+
+        // # Log in as a team admin.
+        cy.apiAdminLogin().then((res) => {
+            sysadmin = res.user;
+        });
+    });
+
+    after(() => {
+        // Login back as admin.
```
Comment (saturninoabril) [inline e2e/cypress/integration/enterprise/guest_accounts/guest_identification_spec.js:43]: Please add `#` or `*` whichever is applicable for consistency. Also in other places.
```suggestion
        // # Login back as admin.
```

File: e2e/cypress/integration/enterprise/guest_accounts/guest_identification_spec.js [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+
+        // # Also ensure that this MFA setting is enforced.
+        cy.findByTestId('ServiceSettings.EnforceMultifactorAuthenticationtrue').check();
+
+        // Click "Save".
+        cy.get('#saveSetting').scrollIntoView().click();
+
+        cy.url().then((url) => {
+            if (url.includes('mfa/setup')) {
+                // complete MFA setup if we are on token setup page /mfa/setup
```
Comment (saturninoabril) [inline e2e/cypress/integration/enterprise/guest_accounts/guest_identification_spec.js:75]: Please start comment with capital letter.

**Review by saturninoabril — CHANGES_REQUESTED** [review]
Thanks @isacikgoz! This is great! Really appreciate your big effort here. I've tested and works as expected. I left minor comments only.

File: e2e/cypress/integration/enterprise/guest_accounts/guest_identification_spec.js (hunk 3 above)
Comment (isacikgoz) [reply e2e/cypress/integration/enterprise/guest_accounts/guest_identification_spec.js:43]: Sure, but some comments are not a step that are defined in the test script. Actually all of the comments without a mark is a test step. So, i think I can add `#` to all of them?

Comment (saturninoabril) [reply e2e/cypress/integration/enterprise/guest_accounts/guest_identification_spec.js:43]: Yes, please. Thanks!

File: e2e/cypress.json [hunk 5]
```diff
@@ -11,7 +11,7 @@
         "adminUsername": "sysadmin",
         "adminPassword": "Sys@dmin-sample1",
         "dbClient": "postgres",
-        "dbConnection": "postgres://mmuser:mostest@localhost/mattermost_test?sslmode=disable\u0026connect_timeout=10",
+        "dbConnection": "postgres://mmuser:mostest@localhost:5432/dba91uot6zhbycbqcu8ain4u8kzh?sslmode=disable\u0026connect_timeout=10",
```
Comment (saturninoabril) [inline e2e/cypress.json:14]: Please revert :)

Comment (isacikgoz) [reply e2e/cypress.json:14]: Aha, sorry :)

Comment (saturninoabril) [reply e2e/cypress.json]: No worries and thanks!

**Review by saturninoabril — APPROVED** [review]

**Review by lindy65 — APPROVED** [review]
Thanks @isacikgoz - looks good to me 👍

**Review by mgdelacroix — APPROVED** [review]
LGTM!! :+1:

**Pull request merged**